### PR TITLE
Switchboard testbench for umi_crossbar (including umi_arbiter and umi_priority)

### PR DESCRIPTION
**Pull request by volkamir**
This PR adds a switchboard testbench and test for testing the crossbar.
The test randomly injects packets to all ports with random destinations and checks their receipt on the output ports. 

The PR also includes the removal of the legacy testbenches.

### CODE REVIEW

File: umi/testbench/test_regif.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    #verilator_run(verilator_bin, plusargs=['trace'])
+    verilator_run(verilator_bin, plusargs=['trace', ('valid_mode', vldmode), ('ready_mode', rdymode)])
+
+    # instantiate TX and RX queues.  note that these can be instantiated without
+    # specifying a URI, in which case the URI can be specified later via the
+    # "init" method
+
+    host = UmiTxRx(host2dut, dut2host)
+
+    print("### Statring test ###")
```
Comment (sgherbst) [inline umi/testbench/test_regif.py:111]: typo

File: umi/testbench/test_crossbar.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    verilator_run(verilator_bin, plusargs=['trace', ('PORTS', ports), ('valid_mode', vldmode), ('ready_mode', rdymode)])
+
+    # instantiate TX and RX queues.  note that these can be instantiated without
+    # specifying a URI, in which case the URI can be specified later via the
+    # "init" method
+
+    umi = [UmiTxRx('', f'rtl2client_{x}.q') for x in range(ports)]
+    tee = [UmiTxRx('', f'tee_{x}.q') for x in range(ports)]
+
+    print("### Statring test ###")
```
Comment (sgherbst) [inline umi/testbench/test_crossbar.py:87]: typo

File: umi/testbench/test_crossbar.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        proc.join()
+
+    for i in range(ports):
+        for j in range(ports):
+            if len(send_queue[i][j]) != len(recv_queue[i][j]):
+                print(f"packets sent: {len(send_queue[i][j])} packets received: {len(recv_queue[i][j])}")
+            assert len(send_queue[i][j]) == len(recv_queue[i][j])
+            for txp, rxp in zip(send_queue[i][j],recv_queue[i][j]):
+                #print(f"{rxp} {txp}")
+                assert txp.cmd == rxp.cmd
```
Comment (sgherbst) [inline umi/testbench/test_crossbar.py:136]: The four asserts can be replaced by `txp == rxp`

**Review by sgherbst — APPROVED** [review]
LGTM!  Was able to run both tests locally.

File: umi/testbench/test_regif.py (hunk 1 above)
Comment (volkamir) [reply umi/testbench/test_regif.py:111]: thanks, fixed.

File: umi/testbench/test_crossbar.py (hunk 2 above)
Comment (volkamir) [reply umi/testbench/test_crossbar.py:87]: fixed

File: umi/testbench/test_crossbar.py (hunk 3 above)
Comment (volkamir) [reply umi/testbench/test_crossbar.py:136]: Correct. This was taken from a different case where some of the fields were manipulated by the dut.
Will replace.

**Review by azaidy — APPROVED** [review]
LGTM!

**Pull request merged**